main.nf

```diff
@@ -0,0 +1,517 @@
+
+Channel.fromPath(file(params.labels_list))
+    .into{labels_list_for_compute;labels_list_for_visualize}
+
+in_opt_metrics = Channel
+    .fromFilePairs("$root/**/metrics/*.nii.gz",
+                    size: -1,
+                    maxDepth:2) {it.parent.parent.name}
+
+in_dwi_data = Channel
+    .fromFilePairs("$root/**/{*dwi.bval,*dwi.bvec,*dwi.nii.gz,*peaks.nii.gz}",
+                    size: 4,
+                    maxDepth:1,
+                    flat: true) {it.parent.name}
+
+(data_for_kernels, data_for_commit) = in_dwi_data
+    .map{sid, bval, bvec, dwi, peaks -> 
+        [tuple(sid, bval, bvec, dwi, peaks),
+        tuple(sid, bval, bvec, dwi, peaks)]}
+    .separate(2)
+
+
+in_t1_labels
+    .join(in_transfo)
+    .set{anat_for_transformation}
+process Transform_T1_Labels {
+    cpus 1
+
+    input:
+    set sid, file(anat), file(labels), file(mat), file(warp) from anat_for_transformation
+
+    output:
+    set sid, "${sid}__labels_warped_int16.nii.gz" into labels_for_transformation, labels_for_decompose
+    set sid, "${sid}__t1_warped.nii.gz" into anat_for_registration, anat_for_concatenate, anat_for_metrics
+
+    script:
+    """
+    antsApplyTransforms -d 3 -i $anat -r $warp -o "${sid}__t1_warped.nii.gz" -t $warp $mat -n Linear
+    antsApplyTransforms -d 3 -i $labels -r $warp -o labels_warped.nii.gz -t $warp $mat -n NearestNeighbor
+    scil_image_math.py convert labels_warped.nii.gz "${sid}__labels_warped_int16.nii.gz" --data_type int16
+    """
+}
+
+
+in_tracking
+    .into{tracking_for_decompose;tracking_for_kernel}
+
+tracking_for_decompose
+    .join(anat_for_concatenate)
+    .join(labels_for_decompose)
+    .set{tracking_labels_for_decompose}
+
+process Decompose_Connectivity {
+    cpus 2
+
+    input:
+    set sid, file(trackings), file(anat), file(labels) from tracking_labels_for_decompose
+
+    output:
+    set sid, "${sid}__decompose.h5" into h5_for_commit, h5_for_skip_commit
+
+    script:
+    """
+    no_pruning_arg=""
+    if $params.no_pruning; then
+        no_pruning_arg="--no_pruning"
+    fi
+    no_remove_loops_arg=""
+    if $params.no_remove_loops; then
+        no_remove_loops_arg="--no_remove_loops"
+    fi
+    no_remove_outliers_arg=""
+    if $params.no_remove_outliers; then
+        no_remove_outliers_arg="--no_remove_outliers"
+    fi
+    if [ `echo $trackings | wc -w` -gt 1 ]; then
+        scil_streamlines_math.py concatenate $trackings tracking_concat.trk --reference $anat --ignore_invalid
+    else
+        mv $trackings tracking_concat.trk
+    fi
+
+    scil_decompose_connectivity.py tracking_concat.trk $labels "${sid}__decompose.h5" --no_remove_curv_dev \
+        \$no_pruning_arg \$no_remove_loops_arg \$no_remove_outliers_arg --min_length $params.min_length \
+        --max_length $params.max_length --loop_max_angle $params.loop_max_angle \
+        --outlier_threshold $params.outlier_threshold
+    """
+}
+
+data_for_kernels
+    .join(tracking_for_kernel)
+    .first()
```

Comment:
Why the first one ?

Reply:
It works with any datasets, it is based on the acquisition scheme.
@mdesco doing it like we talked about would mean you absolutely cannot merge database.
Is it ok like this or we should just compute the kernel everytime?